content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,563 @@
+        """
+            Defines the computation performed at every call.
+        """
+        total_length = padded_sequences_vectors.shape[1]
+        pack_padded_sequences_vectors = pack_padded_sequence(padded_sequences_vectors, lengths, batch_first=True)
+
+        lstm_out, self.hidden_state = self.lstm_network(pack_padded_sequences_vectors)
+        lstm_out, _ = pad_packed_sequence(lstm_out, batch_first=True, total_length=total_length)
+
+        tag_space = self.fully_connected_network(lstm_out)
+        return tag_space.transpose(-1, 1) # we need to transpose since it's a sequence
+
+full_network = FullNetWork(lstm_network, fully_connected_network)
+```
+
+## Summary
+
+So we have created an LSTM network (`lstm_network`), a fully connected network (`fully_connected_network`), those two 
+components are used in the full network. This full network used padded, packed sequences (defined in the forward pass), 
+so we created the `pad_collate_fn` function to process the needed work. The DataLoader will conduct that process. Finally, 
+when we load the data, this will be done using the vectorizer, so the address will be represented using word embeddings. 
+Also, the address components will be converted into categorical value (from 0 to 7).
+
+
+## The Training
+
+Now that we have all the components for the network let's define our Stochastic Gradient Descent 
+([SGD](https://en.wikipedia.org/wiki/Stochastic_gradient_descent)) optimizer.
+
+```python
+optimizer = optim.SGD(full_network.parameters(), lr)
+```
+
+### Poutyne Experiment
+> Disclaimer that I'm a dev on Poutyne, so I will present code using this framework. See the project [here](https://poutyne.org/).
+
+Let's create our experiment using Poutyne for automatically logging in the project root directory (`./`). We will also set
+the loss function and a batch metric (accuracy) to monitor the training.
+
+```python
+exp = Experiment("./", full_network, device=cuda_device, optimizer=optimizer,
+                 loss_function=cross_entropy, batch_metrics=["acc"])
+```
+
+Using our experiment, we can now launch the training as simple as
+
+```python
+exp.train(train_loader, valid_generator=valid_loader, epochs=epoch_number)
+```
+It will take around 6 minutes per epochs, so around an hour for the complete training.
+
+### Results
+The next figure presents the loss and the accuracy during training (blue) and during validation (orange).
+After 10 epochs, we obtain a validation loss and accuracy of `0.01981` and `99.54701` respectively, which is pretty
+good for a first model. Also, we can see that our model did not seem to have overfitted.
+
+![loss_acc](graph/training_graph.png)
+
+## Bigger model
+
+It seems like our model performed pretty well, but just for fun, let's unleash the full potential of LSTM using a
+bidirectional approach (bi-LSTM). What it means is instead of _simply_ using a seeing the sequence from the start to the end, we
```

Comment:
Il a un autre truc qui cloche ici
```suggestion
bidirectional approach (bi-LSTM). What it means is that instead of _simply_ using a seeing the sequence from the start to the end, we
```